File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map= new HashMap<>();
        for (int i=0;i<nums.length;i++){
            int elt= target-nums[i];
            if (map.get(elt)==null){
                map.put(nums[i],i);
            }
            else{
                return map.get(elt)>i? new int[]{i,map.get(elt)}: new int[]{map.get(elt),i};
            }
        };
    
    return null;
    };
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        subset = []

        def backtrack(i, subset):
            if sum(subset) == target:
                return res.append(subset.copy())
            if i>=len(nums) or sum(subset)>target:
                return

            subset.append(nums[i])
            backtrack(i, subset)

            subset.pop()
            backtrack(i + 1, subset)

        backtrack(0, subset)
        return res